peps/pep-0748.rst

```diff
@@ -0,0 +1,1574 @@
+        """
+        A special signaling exception used only when non-blocking or buffer-only I/O is used.
+
+        This error signals that the requested
+        operation cannot complete until more data is written to the network,
+        or until the output buffer is drained.
+
+        This error is should only be raised when it is completely impossible
+        to write any data. If a partial write is achievable then this should
+        not be raised.
+        """
+
+
+    class WantReadError(TLSError):
+        """
+        A special signaling exception used only when non-blocking or buffer-only I/O is used.
+
+        This error signals that the requested
+        operation cannot complete until more data is read from the network, or
+        until more data is available in the input buffer.
+
+        This error should only be raised when it is completely impossible to
+        write any data. If a partial write is achievable then this should not
+        be raised.
+        """
+
+
+    class RaggedEOF(TLSError):
+        """A special signaling exception used when a TLS connection has been
+        closed gracelessly: that is, when a TLS CloseNotify was not received
+        from the peer before the underlying TCP socket reached EOF. This is a
+        so-called "ragged EOF".
+
+        This exception is not guaranteed to be raised in the face of a ragged
+        EOF: some implementations may not be able to detect or report the
+        ragged EOF.
+
+        This exception is not always a problem. Ragged EOFs are a concern only
+        when protocols are vulnerable to length truncation attacks. Any
+        protocol that can detect length truncation attacks at the application
+        layer (e.g. HTTP/1.1 and HTTP/2) is not vulnerable to this kind of
+        attack and so can ignore this exception.
+        """
+
+
+    class ConfigurationError(TLSError):
+        """An special exception that implementations can use when the provided
+        configuration uses features not supported by that implementation."""
+
+
+Certificates
+~~~~~~~~~~~~
+
+This module would define a concrete certificate class. This class would have
+almost no behavior, as the goal of this module is not to provide all possible
+relevant cryptographic functionality that could be provided by X.509
+certificates. Instead, all we need is the ability to signal the source of a
+certificate to a concrete implementation.
+
+For that reason, this certificate class defines three attributes, corresponding
+to the three envisioned constructors: certificates from files, certificates from
+memory, or certificates from arbitrary identifiers. It is possible that
+implementations do not support all of these constructors, and they can
+communicate this to users as described in the “Runtime” section below.
+Certificates from arbitrary identifiers, in particular, are expected to be
+useful primarily to users seeking to build integrations on top of HSMs, TPMs,
```

Comment:
These abbreviations are never defined.

Reply:
The "expand the first use" style should be sufficient for these two (`hardware security modules (HSMs), trusted platform modules (TPMs),` if these are actually the first use)